lib/mini_twitter_client.rb

```diff
@@ -0,0 +1,41 @@
+require 'faraday'
+require 'json'
+
+def connection
+  Faraday.new(
+    url: 'http://localhost:4000/api/tweets',
+    params: {},
+    headers: {
+      'Accept' => 'application/json',
+      'Content-Type' => 'application/json'
+    }
+  )
+end
+
+def tweets
+  response = connection.get
+
+  JSON.parse(response.body)['data'].map do |tweet|
+    {
+      author: tweet['author'],
+      id: tweet['id'],
+      message: tweet['message']
+    }
+  end
+end
+
+def parse_tweets(author)
+  tweets.select { |tweet| tweet[:author].downcase == author.downcase }
+end
+
+def create_tweet(author, message)
+  connection.post { |post| post.body = { "tweet": { "author": author, "message": message } }.to_json }
+end
+
+def del_tweet(id)
+  connection.delete(id)
+end
+
+def del_all_tweets
+  tweets.each { |tweet| del_tweet(tweet[:id]) }
+end
```

Comment:
Looking at `terminal_mini_client.rb` I had to check which method belongs to the Mini Twitter HTTP client and which is defined directly in the terminal mini client. For example there are `del_tweet` and `delete_tweet` methods. It is hard to remember which one is from which part of the app.

Perhaps it is time to start using classes a little bit ;)

For starters, you could move your methods from this file into the `MiniTwitterClient` class:

```ruby
class MiniTwitterClient
  # ....
end
```

Afterwards, in `terminal_mini_client.rb` you could build instance of `MiniTwitterClient` the same way as you are building instance of Faraday:

```ruby
def mini_twitter_client
  MiniTwitterClient.new
end
```

Then it would be clear that a method comes from the `mini_twitter_client`, for example:

```ruby
mini_twitter_client.del_tweet(id)
```

Reply:
Yes, correct, this will sort it out